### PR TITLE
Add license headers

### PR DESCRIPTION
**Pull request by odow**
I'll comment in-line with a few things.

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -24,86 +24,17 @@ The Convex.jl package is licensed under the Simplified "2-clause" BSD License:
...
->LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
->OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
->SOFTWARE.
-
-The files benchmark/pprinthelper.jl, benchmark/pprintjudge.jl,
-benchmark/pprintresult.jl, benchmark/runbenchmarks.jl, and benchmark/runjudge.jl
-and .travis.yml contain code copied from the Transducers.jl package
-(https://github.com/tkf/Transducers.jl/) which is licensed under the MIT License:
-
->Copyright (c) 2018 Takafumi Arakaki
```
Comment (odow) [inline LICENSE.md:56]: I've moved this to a separate LICENSE.md file in `benchmark`. It isn't part of the Convex.jl source code that gets loaded by `using Convex`.

File: LICENSE.md [hunk 2]
```diff
@@ -24,86 +24,17 @@ The Convex.jl package is licensed under the Simplified "2-clause" BSD License:
...
->IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
->FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
->AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
->LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
->OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
->SOFTWARE.
-
-The `TreePrint` module contains code from the AbstractTrees.jl package
-(https://github.com/Keno/AbstractTrees.jl) which is licensed under the
-MIT "Expat" License:
```
Comment (odow) [inline LICENSE.md:78]: I've moved this to the file `src/utilities/tree_prinnt.jl`

File: benchmark/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+>The above copyright notice and this permission notice shall be included in all
+>copies or substantial portions of the Software.
+>
+>THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+>IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+>FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+>AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+>LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+>OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+>SOFTWARE.
```
Comment (odow) [inline benchmark/LICENSE.md:24]: It's not really obvious that code like:
```julia
# This file was copied from Transducers.jl
# which is available under an MIT license (see LICENSE).
using PkgBenchmark
benchmarkpkg(
    dirname(@__DIR__),
    BenchmarkConfig(env = Dict("JULIA_NUM_THREADS" => "1")),
    resultfile = joinpath(@__DIR__, "result.json"),
```
is justified to declare the MIT license. This is just usage of a package, unrelated to Transducers?

File: src/problem_depot/problem_depot.jl [hunk 4]
```diff
@@ -1,5 +1,7 @@
-# Some code in `src/problem_depot` was modified from MathOptInterface
-# which is available under an MIT license (see LICENSE).
```
Comment (odow) [inline src/problem_depot/problem_depot.jl:2]: I've removed the reference to MathOptInterface's MOI. Looking through this file, I don't see anything that justifies it. I don't think using a constant to store a benchmark suite justifies it.

**Review by blegat — APPROVED** [review]

**Comment by ericphanson** [comment]
I think I was probably too conservative with license stuff, added too many. This looks good!

**Pull request merged**